### PR TITLE
Document `MapIdentityApi<TUser>` endpoints

### LINKED ISSUE #31196: Document `MapIdentityApi<TUser>` endpoints
### Description

The first iteration of the .NET 8 version of this article only covered registration and login. We need to document all of the endpoints mapped by `MapIdentityApi<TUser>`.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authentication/identity-api-authorization.md

### Document ID

137d4b94-7b26-3911-b22d-42c754a95fc1

### Article author

@JeremyLikness

---
[Associated WorkItem - 189973](https://dev.azure.com/msft-skilling/Content/_workitems/edit/189973)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #31196 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/authentication/identity-api-authorization.md](https://github.com/dotnet/AspNetCore.Docs/blob/a860dba63cc334c03eff6e90fde3947b15809add/aspnetcore/security/authentication/identity-api-authorization.md) | [How to use Identity to secure a Web API backend for SPAs](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?branch=pr-en-us-31197) |

### CODE REVIEW

**Comment by guardrex** [comment]
BTW @tdykstra ... In case you didn't hear, Jeremy will be sending new logout code shortly.

https://github.com/dotnet/AspNetCore.Docs/issues/31205#issuecomment-1845775832

**Review by Rick-Anderson — APPROVED** [review]

**Comment by guardrex** [comment]
**UPDATE**: Replacement code is being discussed at https://github.com/dotnet/blazor-samples/pull/136/files.

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ This article shows how to use Identity to secure a Web API backend for SPAs such
 
 The steps shown in this article add authentication and authorization to an ASP.NET Core Web API app that:
 
+* Isn't already configured for authentication.
 * Targets `net8.0` or later.
 * Includes OpenAPI support.
```
Comment (JeremyLikness) [inline aspnetcore/security/authentication/identity-api-authorization.md:26]: Not sure we want to make this restriction. Is OpenAPI required, or can we do a callout to the differences if they choose to work on an API _without_ OpenAPI support?

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 2]
```diff
@@ -91,7 +92,7 @@ builder.Services.AddIdentityApiEndpoints<IdentityUser>()
     .AddEntityFrameworkStores<ApplicationDbContext>();
 ```
 
-By default, both cookies and proprietary tokens are activated. Cookies are issued if the `useCookies` query string parameter in the login endpoint is `true`.
+By default, both cookies and proprietary tokens are activated. Cookies are issued at login if the `useCookies` query string parameter in the login endpoint is `true`.
```
Comment (JeremyLikness) [inline aspnetcore/security/authentication/identity-api-authorization.md:95]:
```suggestion
By default, both cookies and proprietary tokens are activated. Cookies and tokens are issued at login if the `useCookies` query string parameter in the login endpoint is `true`.
```

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 3]
```diff
@@ -101,14 +102,14 @@ After the call to `builder.Build()`, call <xref:Microsoft.AspNetCore.Routing.Ide
 app.MapIdentityApi<IdentityUser>();
 ```
 
-## Secure endpoints
+## Secure selected endpoints
```
Comment (JeremyLikness) [inline aspnetcore/security/authentication/identity-api-authorization.md:105]: Should we mention how to secure with a specific claim or permission too? I.e. the difference between:

`RequiresAuthorization()`  

and

`RequiresAuthorization("Admin")`

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 4]
```diff
@@ -208,38 +211,403 @@ A quick way to test authentication is to use the in-memory database and the Swag
...
 
-The tokens are not standard JSON Web Tokens (JWTs). The use of custom tokens is intentional, as the built-in Identity API is meant primarily for simple scenarios. The token option is not intended to be a fully-featured identity service provider or token server, but instead an alternative to the cookie option for clients that can't use cookies.
+The tokens aren't standard JSON Web Tokens (JWTs). The use of custom tokens is intentional, as the built-in Identity API is meant primarily for simple scenarios. The token option isn't intended to be a full-featured identity service provider or token server, but instead an alternative to the cookie option for clients that can't use cookies.
+
+To use token-based authentication, set the `useCookies` query string parameter to `false` when calling the `/login` endpoint. For more information, see [Use the `POST /login` endpoint](#use-the-post-login-endpoint) later in this article.
```
Comment (JeremyLikness) [inline aspnetcore/security/authentication/identity-api-authorization.md:236]:
```suggestion
To use token-based authentication, set the `useCookies` query string parameter to `false` when calling the `/login` endpoint. Tokens use the _bearer_ authentication scheme. Using the token returned from the call to `/login`, subsequent calls to protected endpoints should add the header `Authorization: Bearer <token>` where `<token>` is the access token. For more information, see [Use the `POST /login` endpoint](#use-the-post-login-endpoint) later in this article.
```

**Review by JeremyLikness — APPROVED** [review]
Really great work, appreciate you. I had a few suggestions but overall looks good to me.

**Pull request merged**